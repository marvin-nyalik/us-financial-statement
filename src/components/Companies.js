import { useSelector, useDispatch } from 'react-redux';
import { useEffect } from 'react';
import css from './companies.module.css';
import stock from '../images/statement.jpg';
import Company from './Company';
import Header from './header';
import Showcase from './showcase';
import LoadingSpinner from './Loading';
import { fetchCompanies } from '../redux/companies/companiesSlice';

const Companies = () => {
  const dispatch = useDispatch();
  const loading = useSelector((state) => state.companies.loading);
  const companies = useSelector((state) => state.companies.companies);
  const headline = 'Selected Companies';
  const year = <i className="bx bx-chevron-left-circle" />;
  const showline = `Financial statements of
    10 Selected Companies in the US stock market`;

  useEffect(() => {
    if (companies.length === 0) {
      dispatch(fetchCompanies());
    }
  }, [dispatch, companies.length]);

  if (loading) {
    return (
      <LoadingSpinner />
    );
  }

  return (
    <div>
      <Header back={year} headline={headline} />
      <Showcase image={stock} showline={showline} />
      <div className={css.sbc}>
        <p>STATS BY COMPANY</p>
      </div>
      <div className={css.allCompanies}>
        {companies.map((company) => (
          <div key={company} className={css.cover}>
            <Company name={company} />
          </div>
        ))}
      </div>
    </div>
  );
};

export default Companies;
