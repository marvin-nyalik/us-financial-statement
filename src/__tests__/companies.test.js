import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import Companies from '../components/Companies';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('Companies Component', () => {
  it('matches the snapshot when loading is true', () => {
    const initialState = {
      companies: {
        loading: true,
        companies: [],
        error: null,
      },
    };

    const store = mockStore(initialState);

    const { asFragment } = render(
      <Provider store={store}>
        <Companies />
      </Provider>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
