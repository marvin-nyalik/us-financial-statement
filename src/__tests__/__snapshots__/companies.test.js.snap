// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Companies Component matches the snapshot when loading is true 1`] = `
<DocumentFragment>
  <div
    class="spinner-container"
    data-testid="loading-spinner"
  >
    <div
      class="loading-spinner"
    />
     Loading
  </div>
</DocumentFragment>
`;
